### PR TITLE
Documentation fixes for v0.33

### PR DESCRIPTION
**Pull request by mudit2812**
This PR will host all changes to docs for the v0.33 release.

If you open any PRs that *only* update docs, set this branch as the base, and merge once the documentation check and the RTD build pass. Since it is a doc-only change, you can merge without finishing CI.

Try and remember to cancel the tests in CI to save some runners :)

### CODE REVIEW

**Comment by mudit2812** [comment]
Closing for now. Will reopen once all fixes are pushed to this branch.

**Review by timmysilv — APPROVED** [review]
just qml.device change and romain's transform docs refresh. lgtm!

File: pennylane/__init__.py [hunk 1]
```diff
@@ -165,8 +166,9 @@ def refresh_devices():
 
 # pylint: disable=protected-access
 def device(name, *args, **kwargs):
-    r"""device(name, wires=1, *args, **kwargs)
-    Load a :class:`~.Device` and return the instance.
+    r"""
+    Load a :class:`.Device` (or a new-API :class:`pennylane.devices.Device`)
+    and return the instance.
```
Comment (trbromley) [inline pennylane/__init__.py:171]:
```suggestion
    Load a device and return the instance.
```

**Review by rmoyard — APPROVED** [review]
Thanks :+1: I did a quick check on the device one

**Pull request merged**